src/observe.ts

```diff
@@ -33,7 +34,9 @@ export const disconnectPerfObserversHidden = (): void => {
     poDisconnect(2);
   }
   if (perfObservers[3]) {
-    perfObservers[3].takeRecords();
+    if (typeof perfObservers[3].takeRecords === 'function') {
+      perfObservers[3].takeRecords();
+    }
```

Comment:
I assumed that the calls to `logMetric` and `poDisconnect` should still happen in this scenario, but let me know if that isn't the case.

Reply:
Interesting point, let me think out loud the logic.
Actually more I re-think the logic out loud more I think there is one more bug to solve 🤦🏻‍♂️
The final records are probably never saved in `cls.value`.